src/Avalonia.Controls/Window.cs

```diff
@@ -515,20 +525,13 @@ private void CloseInternal()
             PlatformImpl?.Dispose();
         }
 
-        /// <summary>
-        /// Handles a closing notification from <see cref="IWindowImpl.Closing"/>.
-        /// </summary>
-        protected virtual bool HandleClosing()
+        private bool ShouldCancelClose()
         {
             bool canClose = true;
 
             foreach (var (child, _) in _children.ToList())
             {
-                if (!child.HandleClosing())
-                {
-                    child.CloseInternal();
-                }
-                else
+                if (child.ShouldCancelClose())
```

Comment:
Do we actually want to check all windows even if one already cancelled?

Reply:
I think so, so that you get the closing event?

Reply:
I guess that with your change to pass args in here it makes sense. So you can tell if it will actually close.